Fix build on ARM64EC

AgostonSzepessy opened the pull request:
Fixes build break due to `#pragma intrinsic( _umul128 )` on ARM64EC. This intrinsic only exists on x64. However, ARM64EC builds also define `_M_X64` so the `#pragma intrinsic` needs to be excluded on those.

Commit:
Fix build on ARM64EC

horenmar commented:
I played around with the pragma a bit on compiler explorer, and I think that the best solution is to just remove it completely.

Commit:
Revert "Fix build on ARM64EC"

This reverts commit 7a4f6d73f96f3124108e640005e0559ba8564683.

Commit:
Fix ARM64EC build

Remove `#pragma intrinsic(_umul128)` because it doesn't work on
ARM64EC and x64 works without it.

AgostonSzepessy commented:
I removed the pragma bit.
